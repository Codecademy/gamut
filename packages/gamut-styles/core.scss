@use "core/index";

@forward "utils";
