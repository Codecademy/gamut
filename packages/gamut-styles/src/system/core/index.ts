export * from './registerHandler';
export * from './compose';
export * from './system';
