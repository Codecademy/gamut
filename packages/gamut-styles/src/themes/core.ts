import { createTheme } from '@codecademy/variance';

import {
  borderRadii,
  corePalette,
  elements,
  fontFamily,
  fontSize,
  fontWeight,
  lineHeight,
  mediaQueries,
  spacing,
} from '../variables';

/**
 *  @description  This is the core theme that all other Codecademy themes are extended from
 * and is the only theme that gamut components are typed to. This ensures that gamut components
 * are guaranteed to be interoperable between all contexts, but with differing behaviors.
 */

export const coreTheme = createTheme({
  breakpoints: mediaQueries,
  borderRadii,
  fontSize,
  fontFamily,
  lineHeight,
  fontWeight,
  spacing,
  elements,
})
  .addColors(corePalette)
  .addColorModes('light', {
    light: {
      text: {
        _: 'navy-800',
        accent: 'navy-900',
        disabled: 'navy-500',
        secondary: 'navy-600',
      },
      feedback: {
        error: 'red-600',
        success: 'green-700',
        warning: 'yellow',
      },
      background: {
        _: 'white',
        contrast: 'white',
        current: 'white',
        primary: 'beige',
        selected: 'navy-100',
        disabled: 'navy-200',
        hover: 'navy-200',
      },
      shadow: {
        primary: 'navy-800',
        secondary: 'navy-600',
      },
      primary: {
        _: 'hyper-500',
        hover: 'hyper-400',
        inverse: 'yellow-500',
      },
      secondary: {
        _: 'navy-800',
        hover: 'navy-700',
      },
      danger: {
        _: 'red-500',
        hover: 'red-600',
      },
      interface: {
        _: 'hyper-500',
        hover: 'hyper-400',
      },
      border: {
        primary: 'navy-800',
        secondary: 'navy-600',
        tertiary: 'navy-300',
        disabled: 'navy-500',
      },
    },
    dark: {
      text: {
        _: 'white',
        accent: 'beige',
        disabled: 'white-500',
        secondary: 'white-600',
      },
      feedback: {
        error: 'red-0',
        success: 'green-400',
        warning: 'yellow-0',
      },
      background: {
        _: 'navy-800',
        contrast: 'black',
        current: 'navy-800',
        primary: 'navy-900',
        selected: 'white-100',
        disabled: 'white-200',
        hover: 'white-200',
      },
      shadow: {
        primary: 'white',
        secondary: 'white-600',
      },
      primary: {
        _: 'yellow-500',
        hover: 'yellow-400',
        inverse: 'hyper-500',
      },
      secondary: {
        _: 'white',
        hover: 'white-700',
      },
      danger: {
        _: 'red-0',
        hover: 'red-100',
      },
      interface: {
        _: 'yellow-500',
        hover: 'yellow-400',
      },
      border: {
        primary: 'white',
        secondary: 'white-600',
        tertiary: 'white-300',
        disabled: 'white-500',
      },
    },
  })
  .addScale('borders', ({ colors }: { colors: Record<string, string> }) => ({
    1: `1px solid ${colors['border-primary']}`,
    2: `2px solid ${colors['border-primary']}`,
  }))
  .createScaleVariables('elements')
  .build();

export type CoreThemeShape = typeof coreTheme;

export interface CoreTheme extends CoreThemeShape {}
