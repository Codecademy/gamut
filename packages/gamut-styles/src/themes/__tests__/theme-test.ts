import { coreTheme } from '../core';
import { platformTheme } from '../platform';

describe('themes', () => {
  it.each([
    ['core', coreTheme],
    ['platform', platformTheme],
  ])(`%s - theme shape`, (_, theme) => expect(theme).toMatchSnapshot());
});
