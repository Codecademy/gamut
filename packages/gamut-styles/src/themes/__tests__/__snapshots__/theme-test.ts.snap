// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themes core - theme shape 1`] = `
Object {
  "_getColorValue": [Function],
  "_tokens": Object {
    "colors": Object {
      "beige": "#FFF0E5",
      "beige-100": "#FFF0E5",
      "black": "#000000",
      "blue": "#1557FF",
      "blue-0": "#F5FCFF",
      "blue-300": "#66C4FF",
      "blue-500": "#1557FF",
      "blue-800": "#1D2340",
      "gray-100": "#F5F5F5",
      "gray-200": "#EEEEEE",
      "gray-300": "#E0E0E0",
      "gray-600": "#9E9E9E",
      "gray-800": "#616161",
      "gray-900": "#424242",
      "green": "#008A27",
      "green-0": "#F5FFE3",
      "green-400": "#AEE938",
      "green-700": "#008A27",
      "hyper": "#3A10E5",
      "hyper-400": "#5533FF",
      "hyper-500": "#3A10E5",
      "lightBlue": "#66C4FF",
      "lightGreen": "#AEE938",
      "navy": "#10162F",
      "navy-800": "#10162F",
      "orange": "#FF8C00",
      "orange-500": "#FF8C00",
      "paleBlue": "#F5FCFF",
      "paleGreen": "#F5FFE3",
      "palePink": "#FFF5FF",
      "paleRed": "#DC5879",
      "paleYellow": "#FFFAE5",
      "pink": "#F966FF",
      "pink-0": "#FFF5FF",
      "pink-400": "#F966FF",
      "red": "#E91C11",
      "red-0": "#E85D7F",
      "red-100": "#DC5879",
      "red-400": "#ED4941",
      "red-500": "#E91C11",
      "shadow-black-heavy": "rgba(0,0,0,0.75)",
      "shadow-black-slight": "rgba(0,0,0,0.15)",
      "shadow-white-heavy": "rgba(255,255,255,0.95)",
      "shadow-white-slight": "rgba(255,255,255,0.15)",
      "white": "#ffffff",
      "yellow": "#FFD300",
      "yellow-0": "#FFFAE5",
      "yellow-400": "#CCA900",
      "yellow-500": "#FFD300",
    },
    "elements": Object {
      "headerHeight": Object {
        "base": "4rem",
        "md": "5rem",
      },
      "headerZ": 15,
    },
  },
  "_variables": Object {
    "mode": Object {
      "--color-background": "var(--color-white)",
      "--color-backgroundEmphasized": "var(--color-gray-100)",
      "--color-backgroundMuted": "var(--color-gray-200)",
      "--color-danger": "var(--color-red)",
      "--color-primary": "var(--color-hyper)",
      "--color-primaryDull": "var(--color-hyper-400)",
      "--color-secondary": "var(--color-navy)",
      "--color-secondaryDull": "var(--color-gray-600)",
      "--color-shadow-opaque": "var(--color-shadow-white-heavy)",
      "--color-shadow-solid": "var(--color-gray-200)",
      "--color-success": "var(--color-green)",
      "--color-text": "var(--color-navy)",
      "--color-textMuted": "var(--color-gray-600)",
    },
    "root": Object {
      "--color-beige": "#FFF0E5",
      "--color-beige-100": "#FFF0E5",
      "--color-black": "#000000",
      "--color-blue": "#1557FF",
      "--color-blue-0": "#F5FCFF",
      "--color-blue-300": "#66C4FF",
      "--color-blue-500": "#1557FF",
      "--color-blue-800": "#1D2340",
      "--color-gray-100": "#F5F5F5",
      "--color-gray-200": "#EEEEEE",
      "--color-gray-300": "#E0E0E0",
      "--color-gray-600": "#9E9E9E",
      "--color-gray-800": "#616161",
      "--color-gray-900": "#424242",
      "--color-green": "#008A27",
      "--color-green-0": "#F5FFE3",
      "--color-green-400": "#AEE938",
      "--color-green-700": "#008A27",
      "--color-hyper": "#3A10E5",
      "--color-hyper-400": "#5533FF",
      "--color-hyper-500": "#3A10E5",
      "--color-lightBlue": "#66C4FF",
      "--color-lightGreen": "#AEE938",
      "--color-navy": "#10162F",
      "--color-navy-800": "#10162F",
      "--color-orange": "#FF8C00",
      "--color-orange-500": "#FF8C00",
      "--color-paleBlue": "#F5FCFF",
      "--color-paleGreen": "#F5FFE3",
      "--color-palePink": "#FFF5FF",
      "--color-paleRed": "#DC5879",
      "--color-paleYellow": "#FFFAE5",
      "--color-pink": "#F966FF",
      "--color-pink-0": "#FFF5FF",
      "--color-pink-400": "#F966FF",
      "--color-red": "#E91C11",
      "--color-red-0": "#E85D7F",
      "--color-red-100": "#DC5879",
      "--color-red-400": "#ED4941",
      "--color-red-500": "#E91C11",
      "--color-shadow-black-heavy": "rgba(0,0,0,0.75)",
      "--color-shadow-black-slight": "rgba(0,0,0,0.15)",
      "--color-shadow-white-heavy": "rgba(255,255,255,0.95)",
      "--color-shadow-white-slight": "rgba(255,255,255,0.15)",
      "--color-white": "#ffffff",
      "--color-yellow": "#FFD300",
      "--color-yellow-0": "#FFFAE5",
      "--color-yellow-400": "#CCA900",
      "--color-yellow-500": "#FFD300",
      "--elements-headerHeight": "4rem",
      "--elements-headerZ": 15,
      "@media only screen and (min-width: 1024px)": Object {
        "--elements-headerHeight": "5rem",
      },
      "@media only screen and (min-width: 1200px)": Object {
        "--elements-headerHeight": undefined,
      },
      "@media only screen and (min-width: 1440px)": Object {
        "--elements-headerHeight": undefined,
      },
      "@media only screen and (min-width: 480px)": Object {
        "--elements-headerHeight": undefined,
      },
      "@media only screen and (min-width: 768px)": Object {
        "--elements-headerHeight": undefined,
      },
    },
  },
  "borders": Object {
    "1": "1px solid var(--color-secondary)",
    "2": "2px solid var(--color-secondary)",
  },
  "breakpoints": Object {
    "lg": "@media only screen and (min-width: 1200px)",
    "md": "@media only screen and (min-width: 1024px)",
    "sm": "@media only screen and (min-width: 768px)",
    "xl": "@media only screen and (min-width: 1440px)",
    "xs": "@media only screen and (min-width: 480px)",
  },
  "colors": Object {
    "background": "var(--color-background)",
    "backgroundEmphasized": "var(--color-backgroundEmphasized)",
    "backgroundMuted": "var(--color-backgroundMuted)",
    "beige": "var(--color-beige)",
    "beige-100": "var(--color-beige-100)",
    "black": "var(--color-black)",
    "blue": "var(--color-blue)",
    "blue-0": "var(--color-blue-0)",
    "blue-300": "var(--color-blue-300)",
    "blue-500": "var(--color-blue-500)",
    "blue-800": "var(--color-blue-800)",
    "danger": "var(--color-danger)",
    "gray-100": "var(--color-gray-100)",
    "gray-200": "var(--color-gray-200)",
    "gray-300": "var(--color-gray-300)",
    "gray-600": "var(--color-gray-600)",
    "gray-800": "var(--color-gray-800)",
    "gray-900": "var(--color-gray-900)",
    "green": "var(--color-green)",
    "green-0": "var(--color-green-0)",
    "green-400": "var(--color-green-400)",
    "green-700": "var(--color-green-700)",
    "hyper": "var(--color-hyper)",
    "hyper-400": "var(--color-hyper-400)",
    "hyper-500": "var(--color-hyper-500)",
    "lightBlue": "var(--color-lightBlue)",
    "lightGreen": "var(--color-lightGreen)",
    "navy": "var(--color-navy)",
    "navy-800": "var(--color-navy-800)",
    "orange": "var(--color-orange)",
    "orange-500": "var(--color-orange-500)",
    "paleBlue": "var(--color-paleBlue)",
    "paleGreen": "var(--color-paleGreen)",
    "palePink": "var(--color-palePink)",
    "paleRed": "var(--color-paleRed)",
    "paleYellow": "var(--color-paleYellow)",
    "pink": "var(--color-pink)",
    "pink-0": "var(--color-pink-0)",
    "pink-400": "var(--color-pink-400)",
    "primary": "var(--color-primary)",
    "primaryDull": "var(--color-primaryDull)",
    "red": "var(--color-red)",
    "red-0": "var(--color-red-0)",
    "red-100": "var(--color-red-100)",
    "red-400": "var(--color-red-400)",
    "red-500": "var(--color-red-500)",
    "secondary": "var(--color-secondary)",
    "secondaryDull": "var(--color-secondaryDull)",
    "shadow-black-heavy": "var(--color-shadow-black-heavy)",
    "shadow-black-slight": "var(--color-shadow-black-slight)",
    "shadow-opaque": "var(--color-shadow-opaque)",
    "shadow-solid": "var(--color-shadow-solid)",
    "shadow-white-heavy": "var(--color-shadow-white-heavy)",
    "shadow-white-slight": "var(--color-shadow-white-slight)",
    "success": "var(--color-success)",
    "text": "var(--color-text)",
    "textMuted": "var(--color-textMuted)",
    "white": "var(--color-white)",
    "yellow": "var(--color-yellow)",
    "yellow-0": "var(--color-yellow-0)",
    "yellow-400": "var(--color-yellow-400)",
    "yellow-500": "var(--color-yellow-500)",
  },
  "elements": Object {
    "headerHeight": "var(--elements-headerHeight)",
    "headerZ": "var(--elements-headerZ)",
  },
  "fontFamily": Object {
    "accent": "\\"Suisse\\", \\"Apercu\\", -apple-system, BlinkMacSystemFont,
\\"Segoe UI\\", \\"Roboto\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\",
\\"Helvetica Neue\\", sans-serif",
    "base": "\\"Apercu\\", -apple-system, BlinkMacSystemFont, \\"Segoe UI\\",
\\"Roboto\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\",
sans-serif",
    "monospace": "Monaco, Menlo, \\"Ubuntu Mono\\", \\"Droid Sans Mono\\", Consolas,
monospace",
    "system": "-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Ubuntu\\",
\\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
  },
  "fontSize": Object {
    "14": "0.875rem",
    "16": "1rem",
    "18": "1.125rem",
    "20": "1.25rem",
    "22": "1.375rem",
    "26": "1.625rem",
    "34": "2.125rem",
    "44": "2.75rem",
    "64": "4rem",
  },
  "fontWeight": Object {
    "400": 400,
    "700": 700,
    "base": 400,
    "title": 700,
  },
  "lineHeight": Object {
    "base": 1.5,
    "title": 1.1,
  },
  "mode": "light",
  "modes": Object {
    "dark": Object {
      "background": "navy",
      "backgroundEmphasized": "blue-800",
      "backgroundMuted": "gray-600",
      "danger": "paleRed",
      "primary": "yellow",
      "primaryDull": "yellow-400",
      "secondary": "white",
      "secondaryDull": "gray-200",
      "shadow-opaque": "shadow-black-heavy",
      "shadow-solid": "black",
      "success": "lightGreen",
      "text": "white",
      "textMuted": "gray-200",
    },
    "light": Object {
      "background": "white",
      "backgroundEmphasized": "gray-100",
      "backgroundMuted": "gray-200",
      "danger": "red",
      "primary": "hyper",
      "primaryDull": "hyper-400",
      "secondary": "navy",
      "secondaryDull": "gray-600",
      "shadow-opaque": "shadow-white-heavy",
      "shadow-solid": "gray-200",
      "success": "green",
      "text": "navy",
      "textMuted": "gray-600",
    },
  },
  "spacing": Object {
    "0": 0,
    "12": "0.75rem",
    "16": "1rem",
    "24": "1.5rem",
    "32": "2rem",
    "4": "0.25rem",
    "48": "3rem",
    "64": "4rem",
    "8": "0.5rem",
    "96": "6rem",
  },
}
`;

exports[`themes platform - theme shape 1`] = `
Object {
  "_getColorValue": [Function],
  "_tokens": Object {
    "colors": Object {
      "beige": "#FFF0E5",
      "beige-0": "#FFFBF8",
      "beige-100": "#FFF0E5",
      "black": "#000000",
      "blue": "#1557FF",
      "blue-0": "#F5FCFF",
      "blue-300": "#66C4FF",
      "blue-500": "#1557FF",
      "blue-800": "#1D2340",
      "gold": "#8A7300",
      "gold-800": "#8A7300",
      "gray-100": "#F5F5F5",
      "gray-200": "#EEEEEE",
      "gray-300": "#E0E0E0",
      "gray-600": "#9E9E9E",
      "gray-800": "#616161",
      "gray-900": "#424242",
      "green": "#008A27",
      "green-0": "#F5FFE3",
      "green-400": "#AEE938",
      "green-700": "#008A27",
      "hyper": "#3A10E5",
      "hyper-400": "#5533FF",
      "hyper-500": "#3A10E5",
      "lightBeige": "#FFFBF8",
      "lightBlue": "#66C4FF",
      "lightGreen": "#AEE938",
      "navy": "#10162F",
      "navy-300": "#585C6D",
      "navy-600": "#232940",
      "navy-800": "#10162F",
      "navy-900": "#0B0F21",
      "orange": "#FF8C00",
      "orange-500": "#FF8C00",
      "orange-800": "#D14900",
      "paleBlue": "#F5FCFF",
      "paleGreen": "#F5FFE3",
      "palePink": "#FFF5FF",
      "paleRed": "#DC5879",
      "paleYellow": "#FFFAE5",
      "pink": "#F966FF",
      "pink-0": "#FFF5FF",
      "pink-400": "#F966FF",
      "pink-800": "#CA00D1",
      "purple": "#B3CCFF",
      "purple-300": "#B3CCFF",
      "red": "#E91C11",
      "red-0": "#E85D7F",
      "red-100": "#DC5879",
      "red-400": "#ED4941",
      "red-500": "#E91C11",
      "shadow-black-heavy": "rgba(0,0,0,0.75)",
      "shadow-black-slight": "rgba(0,0,0,0.15)",
      "shadow-white-heavy": "rgba(255,255,255,0.95)",
      "shadow-white-slight": "rgba(255,255,255,0.15)",
      "teal": "#027E97",
      "teal-500": "#027E97",
      "white": "#ffffff",
      "yellow": "#FFD300",
      "yellow-0": "#FFFAE5",
      "yellow-400": "#CCA900",
      "yellow-500": "#FFD300",
    },
    "elements": Object {
      "headerHeight": Object {
        "base": "4rem",
        "md": "5rem",
      },
      "headerZ": 15,
    },
  },
  "_variables": Object {
    "mode": Object {
      "--color-background": "var(--color-navy)",
      "--color-backgroundEmphasized": "var(--color-blue-800)",
      "--color-backgroundMuted": "var(--color-gray-600)",
      "--color-danger": "var(--color-paleRed)",
      "--color-editor-annotation": "var(--color-red-0)",
      "--color-editor-atom": "var(--color-pink-400)",
      "--color-editor-attribute": "var(--color-green-400)",
      "--color-editor-basic": "var(--color-white)",
      "--color-editor-comment": "var(--color-gray-600)",
      "--color-editor-constant": "var(--color-orange-500)",
      "--color-editor-decoration": "var(--color-red-0)",
      "--color-editor-invalid": "var(--color-red-0)",
      "--color-editor-key": "var(--color-blue-300)",
      "--color-editor-keyword": "var(--color-purple-300)",
      "--color-editor-number": "var(--color-red-0)",
      "--color-editor-operator": "var(--color-red-0)",
      "--color-editor-predefined": "var(--color-white)",
      "--color-editor-property": "var(--color-red-0)",
      "--color-editor-regexp": "var(--color-green-400)",
      "--color-editor-string": "var(--color-yellow-500)",
      "--color-editor-tag": "var(--color-red-0)",
      "--color-editor-text": "var(--color-orange-500)",
      "--color-editor-ui-background": "var(--color-navy-900)",
      "--color-editor-ui-indent-active": "var(--color-gray-900)",
      "--color-editor-ui-indent-inactive": "var(--color-gray-800)",
      "--color-editor-ui-line-number": "var(--color-gray-600)",
      "--color-editor-ui-text": "var(--color-white)",
      "--color-editor-value": "var(--color-yellow-500)",
      "--color-editor-variable": "var(--color-green-400)",
      "--color-primary": "var(--color-yellow)",
      "--color-primaryDull": "var(--color-yellow-400)",
      "--color-secondary": "var(--color-white)",
      "--color-secondaryDull": "var(--color-gray-200)",
      "--color-shadow-opaque": "var(--color-shadow-black-heavy)",
      "--color-shadow-solid": "var(--color-black)",
      "--color-success": "var(--color-lightGreen)",
      "--color-text": "var(--color-white)",
      "--color-textMuted": "var(--color-gray-200)",
    },
    "root": Object {
      "--color-beige": "#FFF0E5",
      "--color-beige-0": "#FFFBF8",
      "--color-beige-100": "#FFF0E5",
      "--color-black": "#000000",
      "--color-blue": "#1557FF",
      "--color-blue-0": "#F5FCFF",
      "--color-blue-300": "#66C4FF",
      "--color-blue-500": "#1557FF",
      "--color-blue-800": "#1D2340",
      "--color-gold": "#8A7300",
      "--color-gold-800": "#8A7300",
      "--color-gray-100": "#F5F5F5",
      "--color-gray-200": "#EEEEEE",
      "--color-gray-300": "#E0E0E0",
      "--color-gray-600": "#9E9E9E",
      "--color-gray-800": "#616161",
      "--color-gray-900": "#424242",
      "--color-green": "#008A27",
      "--color-green-0": "#F5FFE3",
      "--color-green-400": "#AEE938",
      "--color-green-700": "#008A27",
      "--color-hyper": "#3A10E5",
      "--color-hyper-400": "#5533FF",
      "--color-hyper-500": "#3A10E5",
      "--color-lightBeige": "#FFFBF8",
      "--color-lightBlue": "#66C4FF",
      "--color-lightGreen": "#AEE938",
      "--color-navy": "#10162F",
      "--color-navy-300": "#585C6D",
      "--color-navy-600": "#232940",
      "--color-navy-800": "#10162F",
      "--color-navy-900": "#0B0F21",
      "--color-orange": "#FF8C00",
      "--color-orange-500": "#FF8C00",
      "--color-orange-800": "#D14900",
      "--color-paleBlue": "#F5FCFF",
      "--color-paleGreen": "#F5FFE3",
      "--color-palePink": "#FFF5FF",
      "--color-paleRed": "#DC5879",
      "--color-paleYellow": "#FFFAE5",
      "--color-pink": "#F966FF",
      "--color-pink-0": "#FFF5FF",
      "--color-pink-400": "#F966FF",
      "--color-pink-800": "#CA00D1",
      "--color-purple": "#B3CCFF",
      "--color-purple-300": "#B3CCFF",
      "--color-red": "#E91C11",
      "--color-red-0": "#E85D7F",
      "--color-red-100": "#DC5879",
      "--color-red-400": "#ED4941",
      "--color-red-500": "#E91C11",
      "--color-shadow-black-heavy": "rgba(0,0,0,0.75)",
      "--color-shadow-black-slight": "rgba(0,0,0,0.15)",
      "--color-shadow-white-heavy": "rgba(255,255,255,0.95)",
      "--color-shadow-white-slight": "rgba(255,255,255,0.15)",
      "--color-teal": "#027E97",
      "--color-teal-500": "#027E97",
      "--color-white": "#ffffff",
      "--color-yellow": "#FFD300",
      "--color-yellow-0": "#FFFAE5",
      "--color-yellow-400": "#CCA900",
      "--color-yellow-500": "#FFD300",
      "--elements-headerHeight": "4rem",
      "--elements-headerZ": 15,
      "@media only screen and (min-width: 1024px)": Object {
        "--elements-headerHeight": "5rem",
      },
      "@media only screen and (min-width: 1200px)": Object {
        "--elements-headerHeight": undefined,
      },
      "@media only screen and (min-width: 1440px)": Object {
        "--elements-headerHeight": undefined,
      },
      "@media only screen and (min-width: 480px)": Object {
        "--elements-headerHeight": undefined,
      },
      "@media only screen and (min-width: 768px)": Object {
        "--elements-headerHeight": undefined,
      },
    },
  },
  "borders": Object {
    "1": "1px solid var(--color-secondary)",
    "2": "2px solid var(--color-secondary)",
  },
  "breakpoints": Object {
    "lg": "@media only screen and (min-width: 1200px)",
    "md": "@media only screen and (min-width: 1024px)",
    "sm": "@media only screen and (min-width: 768px)",
    "xl": "@media only screen and (min-width: 1440px)",
    "xs": "@media only screen and (min-width: 480px)",
  },
  "colors": Object {
    "background": "var(--color-background)",
    "backgroundEmphasized": "var(--color-backgroundEmphasized)",
    "backgroundMuted": "var(--color-backgroundMuted)",
    "beige": "var(--color-beige)",
    "beige-0": "var(--color-beige-0)",
    "beige-100": "var(--color-beige-100)",
    "black": "var(--color-black)",
    "blue": "var(--color-blue)",
    "blue-0": "var(--color-blue-0)",
    "blue-300": "var(--color-blue-300)",
    "blue-500": "var(--color-blue-500)",
    "blue-800": "var(--color-blue-800)",
    "danger": "var(--color-danger)",
    "editor-annotation": "var(--color-editor-annotation)",
    "editor-atom": "var(--color-editor-atom)",
    "editor-attribute": "var(--color-editor-attribute)",
    "editor-basic": "var(--color-editor-basic)",
    "editor-comment": "var(--color-editor-comment)",
    "editor-constant": "var(--color-editor-constant)",
    "editor-decoration": "var(--color-editor-decoration)",
    "editor-invalid": "var(--color-editor-invalid)",
    "editor-key": "var(--color-editor-key)",
    "editor-keyword": "var(--color-editor-keyword)",
    "editor-number": "var(--color-editor-number)",
    "editor-operator": "var(--color-editor-operator)",
    "editor-predefined": "var(--color-editor-predefined)",
    "editor-property": "var(--color-editor-property)",
    "editor-regexp": "var(--color-editor-regexp)",
    "editor-string": "var(--color-editor-string)",
    "editor-tag": "var(--color-editor-tag)",
    "editor-text": "var(--color-editor-text)",
    "editor-ui-background": "var(--color-editor-ui-background)",
    "editor-ui-indent-active": "var(--color-editor-ui-indent-active)",
    "editor-ui-indent-inactive": "var(--color-editor-ui-indent-inactive)",
    "editor-ui-line-number": "var(--color-editor-ui-line-number)",
    "editor-ui-text": "var(--color-editor-ui-text)",
    "editor-value": "var(--color-editor-value)",
    "editor-variable": "var(--color-editor-variable)",
    "gold": "var(--color-gold)",
    "gold-800": "var(--color-gold-800)",
    "gray-100": "var(--color-gray-100)",
    "gray-200": "var(--color-gray-200)",
    "gray-300": "var(--color-gray-300)",
    "gray-600": "var(--color-gray-600)",
    "gray-800": "var(--color-gray-800)",
    "gray-900": "var(--color-gray-900)",
    "green": "var(--color-green)",
    "green-0": "var(--color-green-0)",
    "green-400": "var(--color-green-400)",
    "green-700": "var(--color-green-700)",
    "hyper": "var(--color-hyper)",
    "hyper-400": "var(--color-hyper-400)",
    "hyper-500": "var(--color-hyper-500)",
    "lightBeige": "var(--color-lightBeige)",
    "lightBlue": "var(--color-lightBlue)",
    "lightGreen": "var(--color-lightGreen)",
    "navy": "var(--color-navy)",
    "navy-300": "var(--color-navy-300)",
    "navy-600": "var(--color-navy-600)",
    "navy-800": "var(--color-navy-800)",
    "navy-900": "var(--color-navy-900)",
    "orange": "var(--color-orange)",
    "orange-500": "var(--color-orange-500)",
    "orange-800": "var(--color-orange-800)",
    "paleBlue": "var(--color-paleBlue)",
    "paleGreen": "var(--color-paleGreen)",
    "palePink": "var(--color-palePink)",
    "paleRed": "var(--color-paleRed)",
    "paleYellow": "var(--color-paleYellow)",
    "pink": "var(--color-pink)",
    "pink-0": "var(--color-pink-0)",
    "pink-400": "var(--color-pink-400)",
    "pink-800": "var(--color-pink-800)",
    "primary": "var(--color-primary)",
    "primaryDull": "var(--color-primaryDull)",
    "purple": "var(--color-purple)",
    "purple-300": "var(--color-purple-300)",
    "red": "var(--color-red)",
    "red-0": "var(--color-red-0)",
    "red-100": "var(--color-red-100)",
    "red-400": "var(--color-red-400)",
    "red-500": "var(--color-red-500)",
    "secondary": "var(--color-secondary)",
    "secondaryDull": "var(--color-secondaryDull)",
    "shadow-black-heavy": "var(--color-shadow-black-heavy)",
    "shadow-black-slight": "var(--color-shadow-black-slight)",
    "shadow-opaque": "var(--color-shadow-opaque)",
    "shadow-solid": "var(--color-shadow-solid)",
    "shadow-white-heavy": "var(--color-shadow-white-heavy)",
    "shadow-white-slight": "var(--color-shadow-white-slight)",
    "success": "var(--color-success)",
    "teal": "var(--color-teal)",
    "teal-500": "var(--color-teal-500)",
    "text": "var(--color-text)",
    "textMuted": "var(--color-textMuted)",
    "white": "var(--color-white)",
    "yellow": "var(--color-yellow)",
    "yellow-0": "var(--color-yellow-0)",
    "yellow-400": "var(--color-yellow-400)",
    "yellow-500": "var(--color-yellow-500)",
  },
  "elements": Object {
    "headerHeight": "var(--elements-headerHeight)",
    "headerZ": "var(--elements-headerZ)",
  },
  "fontFamily": Object {
    "accent": "\\"Suisse\\", \\"Apercu\\", -apple-system, BlinkMacSystemFont,
\\"Segoe UI\\", \\"Roboto\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\",
\\"Helvetica Neue\\", sans-serif",
    "base": "\\"Apercu\\", -apple-system, BlinkMacSystemFont, \\"Segoe UI\\",
\\"Roboto\\", \\"Ubuntu\\", \\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\",
sans-serif",
    "monospace": "Monaco, Menlo, \\"Ubuntu Mono\\", \\"Droid Sans Mono\\", Consolas,
monospace",
    "system": "-apple-system, BlinkMacSystemFont, \\"Segoe UI\\", \\"Roboto\\", \\"Ubuntu\\",
\\"Cantarell\\", \\"Fira Sans\\", \\"Droid Sans\\", \\"Helvetica Neue\\", sans-serif",
  },
  "fontSize": Object {
    "14": "0.875rem",
    "16": "1rem",
    "18": "1.125rem",
    "20": "1.25rem",
    "22": "1.375rem",
    "26": "1.625rem",
    "34": "2.125rem",
    "44": "2.75rem",
    "64": "4rem",
  },
  "fontWeight": Object {
    "400": 400,
    "700": 700,
    "base": 400,
    "title": 700,
  },
  "lineHeight": Object {
    "base": 1.5,
    "title": 1.1,
  },
  "mode": "dark",
  "modes": Object {
    "dark": Object {
      "background": "navy",
      "backgroundEmphasized": "blue-800",
      "backgroundMuted": "gray-600",
      "danger": "paleRed",
      "editor-annotation": "red-0",
      "editor-atom": "pink-400",
      "editor-attribute": "green-400",
      "editor-basic": "white",
      "editor-comment": "gray-600",
      "editor-constant": "orange-500",
      "editor-decoration": "red-0",
      "editor-invalid": "red-0",
      "editor-key": "blue-300",
      "editor-keyword": "purple-300",
      "editor-number": "red-0",
      "editor-operator": "red-0",
      "editor-predefined": "white",
      "editor-property": "red-0",
      "editor-regexp": "green-400",
      "editor-string": "yellow-500",
      "editor-tag": "red-0",
      "editor-text": "orange-500",
      "editor-ui-background": "navy-900",
      "editor-ui-indent-active": "gray-900",
      "editor-ui-indent-inactive": "gray-800",
      "editor-ui-line-number": "gray-600",
      "editor-ui-text": "white",
      "editor-value": "yellow-500",
      "editor-variable": "green-400",
      "primary": "yellow",
      "primaryDull": "yellow-400",
      "secondary": "white",
      "secondaryDull": "gray-200",
      "shadow-opaque": "shadow-black-heavy",
      "shadow-solid": "black",
      "success": "lightGreen",
      "text": "white",
      "textMuted": "gray-200",
    },
    "light": Object {
      "background": "white",
      "backgroundEmphasized": "gray-100",
      "backgroundMuted": "gray-200",
      "danger": "red",
      "editor-annotation": "red-500",
      "editor-atom": "pink-800",
      "editor-attribute": "green-700",
      "editor-basic": "navy-800",
      "editor-comment": "navy-300",
      "editor-constant": "orange-800",
      "editor-decoration": "red-500",
      "editor-invalid": "red-500",
      "editor-key": "teal-500",
      "editor-keyword": "blue-500",
      "editor-number": "red-500",
      "editor-operator": "red-500",
      "editor-predefined": "navy-800",
      "editor-property": "red-500",
      "editor-regexp": "green-700",
      "editor-string": "gold-800",
      "editor-tag": "red-500",
      "editor-text": "orange-800",
      "editor-ui-background": "white",
      "editor-ui-indent-active": "navy-900",
      "editor-ui-indent-inactive": "navy-600",
      "editor-ui-line-number": "navy-300",
      "editor-ui-text": "navy-800",
      "editor-value": "gold-800",
      "editor-variable": "green-700",
      "primary": "hyper",
      "primaryDull": "hyper-400",
      "secondary": "navy",
      "secondaryDull": "gray-600",
      "shadow-opaque": "shadow-white-heavy",
      "shadow-solid": "gray-200",
      "success": "green",
      "text": "navy",
      "textMuted": "gray-600",
    },
  },
  "spacing": Object {
    "0": 0,
    "12": "0.75rem",
    "16": "1rem",
    "24": "1.5rem",
    "32": "2rem",
    "4": "0.25rem",
    "48": "3rem",
    "64": "4rem",
    "8": "0.5rem",
    "96": "6rem",
  },
}
`;
